test/e2e/cluster_extension_registryV1_limitations_test.go

```diff
@@ -37,3 +37,29 @@ func TestClusterExtensionPackagesWithWebhooksAreNotAllowed(t *testing.T) {
 		assert.Equal(ct, &ocv1alpha1.BundleMetadata{Name: "package-with-webhooks.1.0.0", Version: "1.0.0"}, clusterExtension.Status.ResolvedBundle)
 	}, pollDuration, pollInterval)
 }
+
+func TestClusterExtensionPackagesWithUnsupportedInstallModesAreNotAllowed(t *testing.T) {
```

Comment:
Is there a way to put this in a unit test instead?

Reply:
@joelanford I started working on this (and another test above `TestClusterExtensionPackagesWithWebhooksAreNotAllowed`) when rukpak was a separate thing running in the cluster with unpacker and all that complicated setup.

The idea was that we have an e2e test which helps us verify current limitations. I was hoping that this test will also help with the helm/rukpak integration into operator-controller. But it got blocked by helm/rukpak integration :)

I do not think that we can test it on `ClusterExtension ` reconciler: rukpak stuff is mocked there.

We can probably test it on [this level](https://github.com/operator-framework/operator-controller/blob/66a063b37f9faa1dae3ff17ad808410023a77f26/internal/handler/registry.go#L26-L40) with a fake FS, but at this point I'm not sure how useful it will be.

Now that we use rukpak as a library - we should probably treat it as such: assume that this stuff if well tested on the rukpak side and do not double test on operator controller.

What do you think?

Reply:
> Now that we use rukpak as a library - we should probably treat it as such: assume that this stuff if well tested on the rukpak side and do not double test on operator controller.

I think with @varshaprasad96 's change in #928, where the actual call to convert the bundle happens fully in rukpak, I agree.